### PR TITLE
Create the first search view and search controller

### PR DESCRIPTION
**Pull request by SunnyXinzeGuo**
1. Wrote a search view that include a form submitting to search method in the controller
2. Add submitting routes and search method in the controller
3. Wrote rspec tests.

### CODE REVIEW

File: app/controllers/my_news_items_controller.rb
Comment (gaoshh711) [inline app/controllers/my_news_items_controller.rb:1]: Emmm, I think it may redirect or redener to the next view

**Review by gaoshh711 — COMMENTED** [review]
Some of commit may require to be checked by Qingcheng. I think there are some overlaps.

**Comment by Clarivy** [comment]
> Some of commit may require to be checked by Qingcheng. I think there are some overlaps.

Yes, I think I need to merge this branch first.

File: app/controllers/my_news_items_controller.rb
Comment (Clarivy) [reply app/controllers/my_news_items_controller.rb:1]: it will becuase there is a search.html.erb in views

**Review by Clarivy — APPROVED** [review]

**Pull request merged**